// react-ui-avatars/lib/index.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _querystring = require('querystring');

var _querystring2 = _interopRequireDefault(_querystring);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function removeUndefinedOrNull(obj) {
  Object.keys(obj).forEach(function (key) {
    if (obj[key] === undefined || obj[key] === null) delete obj[key];
  });

  return obj;
}

function UIAvatar(_ref) {
  var size = _ref.size,
      fontSize = _ref.fontSize,
      length = _ref.length,
      name = _ref.name,
      rounded = _ref.rounded,
      background = _ref.background,
      color = _ref.color,
      uppercase = _ref.uppercase,
      props = _objectWithoutProperties(_ref, ['size', 'fontSize', 'length', 'name', 'rounded', 'background', 'color', 'uppercase']);

  var query = _querystring2.default.stringify(removeUndefinedOrNull({
    size: size,
    'font-size': fontSize,
    length: length,
    name: name,
    rounded: rounded,
    background: background,
    color: color,
    uppercase: uppercase
  }));

  return _react2.default.createElement('img', _extends({ src: 'https://ui-avatars.com/api/?' + query, alt: name }, props));
}
UIAvatar.propTypes = process.env.NODE_ENV !== "production" ? {
  size: _propTypes2.default.number,
  fontSize: _propTypes2.default.number,
  length: _propTypes2.default.number,
  name: _propTypes2.default.string,
  rounded: _propTypes2.default.bool,
  background: _propTypes2.default.string,
  color: _propTypes2.default.string,
  uppercase: _propTypes2.default.boolean
} : {};
UIAvatar.settings = {};
UIAvatar.defaultProps = UIAvatar.settings;

exports.default = UIAvatar;
module.exports = exports['default'];
